src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -865,8 +865,7 @@ private List<DataProviderMetadata> GetDataProvidersMetadata(DataProviders.DataPr
             {
                 if (dataProvider.FieldType.IsInterface)
                 {
-                    var metadataProvider = dataProvider.GetValue(dataProviders) as IMetadataProvider;
-                    if (metadataProvider != null)
+                    if (dataProvider.GetValue(dataProviders) is IMetadataProvider metadataProvider)
```

Comment:
> Since MDM is a field of type Func<MdmDataSource, IMdmDataProvider> in DataProviders class, our default logic of casting the field to Interface is not working and we are not getting the dataProvider metadata.

Are you saying this never used to work?